feat: Migrate from Restana to Express, add large data support and misc changes (#381)
dapr/js-sdk: daprdocs/content/en/js-sdk-docs/js-server/_index.md

> ℹ️ **Note:** The `app-port` is required here, as this is where our server will need to bind to. Dapr will check for the application to bind to this port, before finishing start-up.

### Using HTTP (bring your own express webserver)

> By default, for sidecar to application communication we bring a webserver that is configured to listen to the Sidecar calls as defined in the Dapr API specification. However, in some scenarios (e.g., when you are building a REST API backend and want to integrate Dapr directly in it) you want to bring your own webserver.
> 
> As the Dapr JS SDK team, we do support this scenario, but currently this is only available for [`express`](https://www.npmjs.com/package/express).

Reviewer: ```suggestion
Instead of using the built-in web server for Dapr sidecar to application communication, you can also bring your own instance. This is helpful in scenarios like when you are building a REST API back-end and want to integrate Dapr directly in it.  

Note, this is currently available for [`express`](https://www.npmjs.com/package/express) only.
```

> > 💡 Note: when bringing your own webserver, we will introduce certain overrides that might happen (e.g., configuring the body size, adding certain routes, ...). We avoid collisions in routes by having our own uniques ones, but cannot guarantee collissions.

Reviewer: ```suggestion
> 💡 Note: when using a custom web-server, the SDK will configure server properties like max body size, and add new routes to it. The routes are unique on their own to avoid any collisions with your application, but it's not guaranteed to not collide.
```

1. Should we configure the body size if user brings their own server? That option should be ignored in this case IMO.
2. Should we customize Dapr routes with a specific prefix? for example - `/dapr-server-<route>`, to ensure zero collisions?

Author: Docs adaptation: OK

1. I think we should, we should still have tests passing and we require certain middleware for certain body types. Else we depend on the user to handle cloudevents for example
2. For now I would keep AS-IS, let's see if users require it first seeing the high impact of this

Reviewer: I was just wondering if configuring max body limit on user's express instance will be intrusive, but I see your point. We should wait for some feedback on this feature and take a call I guess!

```typescript
import { DaprServer, CommunicationProtocolEnum } from "@dapr/dapr";
import express from "express";

> const yourApp = express();

Reviewer: ```suggestion
const myApp = express();
```

> yourApp.get("/my-custom-endpoint", (req, res) => {

Reviewer: ```suggestion
myApp.get("/my-custom-endpoint", (req, res) => {
```

  res.send({ msg: "My own express app!" });
});

const daprServer = new DaprServer(
  "127.0.0.1", // App Host
  "50002", // App Port
  daprHost,
  daprPort,
  CommunicationProtocolEnum.HTTP,
  {},
  {

>     serverHttp: yourApp,

Reviewer: ```suggestion
    serverHttp: myApp,
```

  },
);

> // initialize subscribtions, ... before server start
> // the dapr sidecar relies on these
> // this will also initialize the app server itself (removing the need for app.listen to be called)

Reviewer: ```suggestion
// Initialize subscriptions before the server starts, the Dapr sidecar uses it.
// This will also initialize the app server itself (removing the need for `app.listen` to be called).
```
